android/filament-android/src/main/java/com/google/android/filament/View.java

```diff
@@ -582,6 +583,46 @@
         VSM
     }
 
+    /**
+     * View-level options for Shadowing.
+     *
+     * @see View#setShadowOptions
+     */
+    public static class ShadowOptions {
+        /**
+         * The shadow mapping technique this View uses.
+         *
+         * The ShadowType affects all the shadows seen within the View.
+         *
+         * <p>
+         * {@link ShadowType#VSM} imposes a restriction on marking renderables as only shadow
+         * receivers (but not casters). To ensure correct shadowing with VSM, all shadow participant
+         * renderables should be marked as both receivers and casters. Objects that are guaranteed
+         * to not cast shadows on themselves or other objects (such as flat ground planes) can be
+         * set to not cast shadows, which might improve shadow quality.
+         * </p>
+         *
+         * <strong>Warning: This API is still experimental and subject to change.</strong>
+         */
+        public ShadowType shadowType;
+
+        /**
+         * Sets the number of anisotropic samples to use when sampling a VSM shadow map. If greater
+         * than 0, mipmaps will automatically be generated each frame for all lights.
+         *
+         * <p>
+         * The number of anisotropic samples = 2 ^ vsmAnisotropyLog2.
+         * </p>
+         *
+         * <p>
+         * Only applicable when shadowType is set to ShadowType::VSM.
+         * </p>
+         *
+         * <strong>Warning: This API is still experimental and subject to change.</strong>
+         */
+        public int vsmAnisotropyLog2;
```

Comment:
Should we have a separate struct for VSM options? I imagine there could be others (MSAA for instance). Also just call the parameter vsmAnisotropy.

Reply:
Hm, yeah that might make more sense. I was trying to avoid having structs with only a single option in them. MSAA is set on a per-light basis, but I can see needing to add "blur amount" in the future.

Reply:
So in that case, I'll revert the removal of `setShadowType` and rename this struct to `VsmOptions`.